### PR TITLE
More complete example on how to use the XING API in PHP

### PR DESCRIPTION
**Pull request by dawez**
- Implemented more functions using the new functions from the current [xing-hybridauth](https://github.com/FabianBeiner/HybridAuth-XING)
- Using the new (non deprecated) image formats
- Display user contacts 
- Search for jobs / emails
- Added TWIG templates to render the pages
- Refactored the code in separated files

### CODE REVIEW

**Comment by JanAhrens** [comment]
Hi @dawez,

Thanks for your contribution. I'll take some time on the weekend to review it. Unfortunately I'm busy before that. Sorry for the delay.

Cheers, Jan

File: php/app/routes/home.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+            $currentUser = $xingProvider->getUserProfile();
+
+            $viewVariablesArray = array_merge( $viewVariablesArray, array(
+                'currentPage' => 'home',
+                'profile' => array(
+                    'displayName' => $currentUser->displayName,
+                    'photoURL' => $currentUser->photoURL
+                ),
+                'isLoggedIn' => $hybridauth->isConnectedWith( 'XING' ),
+                'contacts' => $xingProvider->getUserContacts( XingUserPicureSize::SIZE_64X64 )
```
Comment (JanAhrens) [inline php/app/routes/home.php]: When I tested this PR, I got the following error message:
```
PHP Fatal error:  Class 'XingUserPicureSize' not found in /vagrant/app/routes/home.php on line 38
```

Is seems that [XingUserPictureSize](https://github.com/FabianBeiner/HybridAuth-XING/blob/master/Hybrid/Providers/XINGUserPictureSize.php) from HybridAuth-XING isn't loaded here.

@dawez: Unfortunately I've no experience with the Composer load mechanism. Can you please fix this appropriately?

Comment (JanAhrens) [inline php/app/routes/home.php]: I guess the problem is that Composer is choosing a very old version of the hybridauth package. 

Here's a snippet from my `composer.lock` file. The referenced version (hybridauth/hybridauth@6d89473b379fc7a904e69ee63f548af3308bd4d9) is from 2013. 

Maybe the [`dev-master` reference](https://github.com/xing/xing-api-samples/blob/b396ada7cd3ff97f5cfbb69755158ff41dd23ec3/php/composer.json#L4) isn't working properly? On packagist.org, the [dev-master snapshot](https://packagist.org/packages/hybridauth/hybridauth#dev-master) is from 2015-10-01 12:26 UTC. This should include the newest release.
```
    "packages": [
        {
            "name": "hybridauth/hybridauth",
            "version": "dev-master",
            "source": {
                "type": "git",
                "url": "https://github.com/hybridauth/hybridauth.git",
                "reference": "6d89473b379fc7a904e69ee63f548af3308bd4d9"
            }
```

@dawez: Do you have an idea why Composer is choosing the old hybridauth version? Can you reproduce this?

Comment (JanAhrens) [inline php/app/routes/home.php]: @dawez: I was able to solve this problem myself. The issue was, that I still had an old `vendor/`library in my local clone. Unfortunately Composer didn't check for newer version and used the local and old one instead. After I deleted the folder and ran `php composer.phar install` again, it worked.

Sorry for the noise!

**Comment by JanAhrens** [comment]
While testing the contacts section, I found and reported a bug in HybridAuth-XING. See https://github.com/FabianBeiner/HybridAuth-XING/issues/11 for details. I think it's wise to wait for a fix, before merging this PR.

**Comment by JanAhrens** [comment]
The layout of the job search results is broken in Firefox 39.0 on Linux. The problem appears when the width of the browser window gets wider than 1190px.

broken-layout

**Comment by dawez** [comment]
Layout issues fixed.

**Comment by JanAhrens** [comment]
Thanks @dawez. It looks really awesome!

I manually merged it: 8d416d29562d3bf754b1cfd52fff59dbf5bd65f8

**Pull request closed without merging**